worker-document/src/main/java/com/hpe/caf/worker/document/impl/ScriptImpl.java

```diff
@@ -237,6 +263,31 @@ public void raiseEvent(final String event, final Object... args)
 
         // Check if there is a JavaScript function event handler for the event
         final Object eventHandler = bindings.get(event);
+        if (scriptSpec.getEngineType() == ScriptEngineType.GRAAL_JS) {
+            graalHandleEvent(eventHandler, args);
+        } else {
+            nashornHandleEvent(eventHandler, args);
+        }
+    }
+
+    private static void graalHandleEvent(final Object eventHandler, final Object[] args)
+    {
+        if (!(eventHandler instanceof Function)) {
+            return;
+        }
+
+        final Value jsEventHandler = Value.asValue(eventHandler);
```

Comment:
Does this definitely hook back into the correct context?  I looked at the JavaDoc but I couldn't tell for sure.  Did you step through it in the debugger to make sure?